x/twap/types/keys.go

```diff
@@ -59,11 +54,9 @@ func FormatMostRecentTWAPKey(poolId uint64, denom1, denom2 string) []byte {
 	return []byte(fmt.Sprintf("%s%s%s%s%s%s", mostRecentTWAPsPrefix, poolIdS, KeySeparator, denom1, KeySeparator, denom2))
 }
 
-// TODO: Replace historical management with ORM, we currently accept 2x write amplification right now.
-func FormatHistoricalTimeIndexTWAPKey(accumulatorWriteTime time.Time, poolId uint64, denom1, denom2 string) []byte {
+func FormatHistoricalPoolIndexDenomPairTWAPKey(poolId uint64, denom1, denom2 string) []byte {
 	var buffer bytes.Buffer
-	timeS := osmoutils.FormatTimeString(accumulatorWriteTime)
-	fmt.Fprintf(&buffer, "%s%s%s%d%s%s%s%s", HistoricalTWAPTimeIndexPrefix, timeS, KeySeparator, poolId, KeySeparator, denom1, KeySeparator, denom2)
+	fmt.Fprintf(&buffer, "%s%d%s%s%s%s", HistoricalTWAPPoolIndexPrefix, poolId, KeySeparator, denom1, KeySeparator, denom2)
```

Comment:
This needs to include the final KeySeparator!

Reply:
Can you explain why? The previous format did not have the final KeySeparator either

Reply:
The prior one was a fixed length prefix, this one is not. So for a pool with multiple denims, you can imagine edge cases where iterators go into other denoms for deleting (not respecting the preserve latest twap condition)

Reply:
The prior one wasn't a prefix It was the full key

Reply:
In general, for anything that involves prefixes (not full key) it _must_ include the final key separator, else iterations may include things that are just longer in the final key segment. (e.g. longer denom 1's) which could happen in pools with > 2 toknes.

Reply:
Dang, we should have a CI for this or something

Reply:
Ideally we switch to collections / ORM, and quit constructing keys ourselves :)